src/Microsoft.AspNetCore.JsonPatch/Internal/PocoAdapter.cs

```diff
@@ -160,7 +161,7 @@ public class PocoAdapter : IAdapter
             }
 
             var currentValue = jsonProperty.ValueProvider.GetValue(target);
-            if (!Equals(currentValue, convertedValue))
```

Comment:
Let's make sure we have good tests for some interesting cases:
- string
- number
- object
- array

Reply:
The unit tests for all the adapters cover these. I am adding integration tests as part of the clean up for #105